src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionsConfigExtensions.cs

```diff
@@ -1,25 +1,29 @@
 namespace NServiceBus
 {
     using System;
+    using System.Reflection;
     using Pipeline;
+    using Routing;
+    using Routing.MessageDrivenSubscriptions;
     using Settings;
+    using Transports;
 
     /// <summary>
     /// Provides extensions for configuring message driven subscriptions.
     /// </summary>
     public static class MessageDrivenSubscriptionsConfigExtensions
     {
         /// <summary>
-        /// Set a Authorizer to be used when verifying a <see cref="MessageIntentEnum.Subscribe" /> or
+        /// Sets an authorizer to be used when verifying a <see cref="MessageIntentEnum.Subscribe" /> or
         /// <see cref="MessageIntentEnum.Unsubscribe" /> message.
         /// </summary>
         /// <remarks>
-        /// This is a "single instance" extension point. So calling this api multiple time will result in only the last
+        /// This is a "single instance" extension point, so calling this api multiple times will result in only the last
         /// one added being executed at message receive time.
         /// </remarks>
-        /// <param name="transportExtensions">The <see cref="TransportExtensions" /> to extend.</param>
+        /// <param name="transportExtensions">The <see cref="TransportExtensions&lt;T&gt;" /> to extend.</param>
         /// <param name="authorizer">The <see cref="Func{TI,TR}" /> to execute.</param>
-        public static void SubscriptionAuthorizer(this TransportExtensions transportExtensions, Func<IIncomingPhysicalMessageContext, bool> authorizer)
+        public static void SubscriptionAuthorizer<T>(this TransportExtensions<T> transportExtensions, Func<IIncomingPhysicalMessageContext, bool> authorizer) where T : TransportDefinition, IMessageDrivenSubscriptionTransport
```

Comment:
do we need to obsolete the old API? Seems like it doesn't make much sense since you shouldn't be using this API if the transport definition doesn't implement `IMessageDrivenSubscriptionTransport` anyway. But an obsolete message in that case wouldn't hurt. But do we have a problem with ambiguous signatures in case the transport implements `IMessageDrivenSubscriptionTransport`? (also we would keep the noise on the config although the method is obsolete)